app/models/discourse_activity_pub_object.rb

```diff
@@ -61,14 +57,19 @@ def self.handle_model_callback(model, ap_type_sym)
 
       object.save!
 
-      if ap_type_sym != :update
-        DiscourseActivityPubActivity.create!(
+      unless ap_type_sym == :update && !model.activity_pub_published?
+        activity_attrs = {
           local: true,
           actor_id: model.activity_pub_actor.id,
           object_id: object.id,
           object_type: 'DiscourseActivityPubObject',
           ap_type: ap.type
+        }
+        unless DiscourseActivityPubActivity.exists?(
```

Comment:
I'm not entirely sure what this does either.

Reply:
This effectively batches Update activities. Consider the following:

1. I edit a post in Discourse at 8:35 am
2. This schedules an Update activity to be published at 8:40am
3. I edit the same post again at 8:36am

Rather than scheduling two different Update activities, we just keep the originally scheduled activity and update the Object associated with that activity (i.e. the Note with the updated `content`). There is a spec describing this case here https://github.com/discourse/discourse-activity-pub/pull/14/files#diff-56aba1a7764ab54eceaa21fbb44a497c6a8d0bbdfafc7886ba87b5e30091efe8R155

Reply:
That makes sense, thanks for the explanation.